src/platform/intel/cavs/lib/pm_runtime.c

```diff
@@ -328,8 +328,8 @@ static inline void cavs_pm_runtime_core_dis_memory(uint32_t index)
 	/* Address is calculated for index (0 for the primary core) minus one
 	 * since _sof_core_s_start is first secondary core stack address
 	 */
-	core_memory_ptr = (char *)&_sof_core_s_start
-		+ (index - 1) * SOF_CORE_S_SIZE;
+	core_memory_ptr = cache_to_uncache((char *)&_sof_core_s_start
+		+ (index - 1) * SOF_CORE_S_SIZE);
```

Comment:
probably safer to 
```c
if (is_cache(addr))
    addr = cache_to_uncache(addr);
```
This would allow clients to pass either flavour.

Reply:
`cache_to_uncache()` and `uncache_to_cache()` use bit operations now, so they're safe to apply multiple times.

Reply:
Don't understand, this doesn't look right to me. The call to `cavs_pm_memory_hp_sram_power_gate()` below calculates memory bank numbers by comparing the address, passed from this function to `HP_SRAM_BASE` and the latter is a cached address, e.g. on TGL
```
#define HP_SRAM_BASE		0xBE000000
```
Am I missing something?

Reply:
yes, @slawblauciak @bkokoszx this looks not quite reasonable to me neither, looks is the change actually don't power down any of the banks(the uncached address not in the range at all so 'mask's are always 0) rescue us from the panic?